swap_meet/vendor.py

```diff
@@ -1,2 +1,79 @@
+from swap_meet.item import Item
+
 class Vendor:
-    pass
+    
+    def __init__(self, inventory=[]):
+        self.inventory = inventory 
+
+    def add(self, item):
+        #self.item = item
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        #self.item = item
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item 
+        else: 
+            return False
+
+    def get_by_category(self, category):
+        list_of_categories = []
+        for item in self.inventory:
+            if item.category == category:
+                list_of_categories.append(item)
+        return list_of_categories
+
+    def swap_items(self, other_vendor, my_item, their_item): 
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        else:
+            self.inventory.remove(my_item)
+            other_vendor.inventory.append(my_item)
+            other_vendor.remove(their_item)
+            self.inventory.append(their_item)
+            return True
+    
+    def swap_first_item(self, friendor):
+        # my_first_item = self.inventory[0]
+        # their_first_item = friendor.inventory[0]
+        #for item in self.inventory:
+        #if self.inventory or friendor.inventory == [""]:
+            #return False
+        #else:
+        if len(self.inventory) != 0 and len(friendor.inventory) != 0:
+            friendor.inventory.append(self.inventory[0])
+            self.inventory.remove(self.inventory[0])
+            self.inventory.append(friendor.inventory[0])
+            friendor.remove(friendor.inventory[0])
```

Comment:
hmm haven't we seen this before? why not use `swap_items` helper function here and pass in the first item of the lists